# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoEncoder.cpp
# Commit: 53c348265bb8
# Full Hash: 53c348265bb848d7b087b700a641ae614b61eb0b
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Validate H264 profile right after parsing. r=chunmin
#   
#   This would error out later anyways, but causes an assertion in ASAN build
#   because the integer isn't part of the enum.
#   
# ==============================================================================

diff -r f55b735f283a -r 53c348265bb8 dom/media/webcodecs/VideoEncoder.cpp
--- a/dom/media/webcodecs/VideoEncoder.cpp	Fri Dec 22 21:39:22 2023 +0000
+++ b/dom/media/webcodecs/VideoEncoder.cpp	Fri Dec 22 21:39:22 2023 +0000
@@ -234,8 +234,13 @@
     } else {
       format = H264BitStreamFormat::AVC;
     }
-    ExtractH264CodecDetails(mCodec, profile, constraints, level);
-    specific.emplace(H264Specific(static_cast<H264_PROFILE>(profile), format));
+    if (ExtractH264CodecDetails(mCodec, profile, constraints, level)) {
+      if (profile == H264_PROFILE_BASE || profile == H264_PROFILE_MAIN ||
+          profile == H264_PROFILE_EXTENDED || profile == H264_PROFILE_HIGH) {
+        specific.emplace(
+            H264Specific(static_cast<H264_PROFILE>(profile), static_cast<H264_LEVEL>(level), format));
+      }
+    }
   }
   // Only for vp9, not vp8
   if (codecType == CodecType::VP9) {
